tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/config/ITmfConfigurationSource.java

```diff
@@ -43,6 +46,31 @@ public interface ITmfConfigurationSource {
      */
     ITmfConfiguration create(Map<String, Object> parameters) throws TmfConfigurationException;
 
+    /**
+     * Creates a new configuration instance.
+     * <p>
+     * The parameters to be provided are described by
+     * {@link ITmfConfigurationSourceType#getConfigParamDescriptors()}.
+     *
+     * @param configuration
+     *            The input configuration parameters to create a configuration instance.
+     * @return a new {@link ITmfConfiguration} if successful
+     * @throws TmfConfigurationException
+     *             If the creation of the configuration fails
+     * @since 9.5
+     */
+    @SuppressWarnings("null")
+    default ITmfConfiguration create(ITmfConfiguration configuration) throws TmfConfigurationException {
+        Map<String, Object> map = new HashMap<>();
+        try {
+            String json = new Gson().toJson(configuration.getParameters());
+            map = new Gson().fromJson(json, Map.class);
+        } catch (Exception e) {
+            throw new TmfConfigurationException("Can't convert input configuration to Map to create configuration", e); //$NON-NLS-1$
+        }
+        return create(map);
```

Comment:
You can just pass configuration.getParameters() directly?

Reply:
True. That will make it simpler and not dependency to GSON is added.